src/Log/LogService.php

```diff
@@ -163,7 +163,11 @@ public function createLog($name, $priority = null, $format = null, $filename = n
         $level = $this->convertPriorityFromString($priority);
 
         if ($format === null) {
-            $format = $logConfig->format;
+            if (isset($logConfig->logging->log->$name->format)) {
```

Comment:
Why duplicate the job of `getLogConfig` here?

Reply:
I assumed `merge()` function would just add the config to the previous config. I saw now that it overrides the same-named configs. So will remove it now.